src/transaction/log_tran_table.c

```diff
@@ -1588,6 +1588,14 @@ logtb_clear_tdes (THREAD_ENTRY * thread_p, LOG_TDES * tdes)
   LSA_SET_NULL (&tdes->rcv.atomic_sysop_start_lsa);
   LSA_SET_NULL (&tdes->rcv.analysis_last_aborted_sysop_lsa);
   LSA_SET_NULL (&tdes->rcv.analysis_last_aborted_sysop_start_lsa);
+
+  /* reflect the session (or prm_Def) */
+  tdes->isolation = (TRAN_ISOLATION) prm_get_integer_value (PRM_ID_LOG_ISOLATION_LEVEL);
+  tdes->wait_msecs = prm_get_integer_value (PRM_ID_LK_TIMEOUT);
+  if (tdes->wait_msecs > 0)
+    {
+      tdes->wait_msecs = tdes->wait_msecs * 1000;
```

Comment:
@Yechan0815 The manual say that 'the max value of lock_timeout is INT_MAX'. But there is no upper limit of this system parameter. And the type of tdes->wait_msecs is INT. If user set the lock_timeout=INT_MAX, then the behavior of lock_timeout work unexpectedly. Do you consider it?

Reply:
@hornetmj 

Unfortunately, it was not considered.....

While processing this issue, I didn't see any exception handling related to `set transaction`, `set system parameters`. Therefore, it was not considered extra.
But, both the 'set transaction' and the modified 'set system parameters' have an error.

Reply:
@Yechan0815 I think there is other choice for it. The lower or upper bound check is executed on each client or server side when the value is set. And you make that user can set the max value of the lock_timeout to the INT_MAX/1000(but, I think it is not a good choice). One of choices is that you save the the value of the lock_timeout as seconds. And the value will be converted to the msec when used.

There may exists other options for you. The purpose of this issue is that the user feel that the change of some parameter works well including lower and upper limitation. If you think you need to error handling before this is done, then you just share your idea through PR or jira issue or Teams(internal) and make an issue for it. And it will block this issue.

Reply:
@hornetmj 

It took me a while to check some.

For system parameters, the server does not handle an exception separately after the client handles an exception for the value. So, it seems like we need to set upper of the value.
Also, it seems that a condition should be added to the `check_timeout_value()` function that checks the range of values at `set transaction timeout`.

Then, there is not much to modify, I think it can be fixed in this issue.

what do you think about this?

Reply:
The default unit of `lock_timeout` is "ms", and it seems to be converted to in "sec" unit  by `prm_msec_to_sec()` internally. It means that users can set it "INT_MAX/1000 seconds" at most and in turn, this conversion causes no overflow issue. `lock_timeout_secs` may have this issue, but this is deprecated. I'm not sure how it works if a user sets it to "INT_MAXs" or "INT_MAXm", though.

Reply:
@Yechan0815 Please go a head. And please check the feedback from @wrlawodms.

Reply:
@wrlawodms 

'lock_timeout_secs' seems to be deprecated.

I couldn't find the part where `PRM_ID_LK_TIMEOUT` changes the value of `PRM_ID_LK_TIMEOUT_SECS`.
`set_dup` (`prm_msec_to_sec ( ... )`) and `get_dup` (`prm_sec_to_msec ( ... )`) of `PRM_ID_LK_TIMEOUT` were also not called when sysprm set/get occurs.
And when I check, the value of `PRM_ID_LK_TIMEOUT_SECS` is not actually reflected and it was still -1 (INFINITE).

Also, `PRM_ID_LK_TIMEOUT` is stored in `sec` even though `PRM_ID_LK_TIMEOUT_SECS` exists.

Reply:
@wrlawodms 

If a value over INT_MAX is input, overflow occurs and the value is stored as it is.
Therefore, I think it is better to set the upper limit about it.

Reply:
When I set `lock_timeout` to INT_MAX + 1 (2147483647+1), `prm_msec_to_sec()` is called and 2147484 is stored. I tested it on the Standalone mode, though. I guess it works on the CS mode in the same way because sysprm_generate_new_value()` is called too on the CS mode. 

```
#0  prm_msec_to_sec (out_val=0x7fffffff6018, out_type=PRM_INTEGER, in_val=0x7fffffff6010, in_type=PRM_BIGINT) at /home/cubrid/cubrid/dev/src/base/system_parameter.c:7493
#1  0x00007ffff5c95e2f in prm_check_range (prm=0x7ffff6c7bed0 <prm_Def+2288>, value=0x7fffffff6098) at /home/cubrid/cubrid/dev/src/base/system_parameter.c:9131
#2  0x00007ffff5c9654c in sysprm_generate_new_value (prm=0x7ffff6c7bed0 <prm_Def+2288>, value=0x7fffffff61ad "400000", check=true, new_value=0x4e817f8)
    at /home/cubrid/cubrid/dev/src/base/system_parameter.c:9435
#3  0x00007ffff5c92f37 in sysprm_validate_change_parameters (data=0x7fffffff6a30 "lock_timeout=400000", check=true, assignments_ptr=0x7fffffff69f0)
    at /home/cubrid/cubrid/dev/src/base/system_parameter.c:7897
#4  0x00007ffff5bc2669 in db_set_system_parameters (data=0x7fffffff6a30 "lock_timeout=400000") at /home/cubrid/cubrid/dev/src/compat/db_admin.c:2718
#5  0x00007ffff5b7ce7e in csql_set_sys_param (arg_str=0x4e95005 "lock_timeout=400000") at /home/cubrid/cubrid/dev/src/executables/csql.c:2348
```

Reply:
@wrlawodms 

I made a mistake in the description. Sorry for the confusion.

When `lock_timeout` is set to `2147484000`, the value is stored as `2147484`. This value does not exceed INT_MAX now, but when it is multiplied by 1000 to apply to `tdes` later, overflow occurs and the overflowed value is applied.

So an upper limit was needed.

Reply:
Oh, I misunderstood the issue. Thank you for elaborating.

> (Do you know why is set system parameters lock_timeout inputted in msec and stored in sec, and set transaction inputed in sec and stored in msec? This is just a curiosity.)

I don't know exactly. I guess the developer wanted to share the same variable with `lock_timeout_in_secs`, which is deprecated now and in the second unit. Maybe for the compatibility.